src/main/java/org/vaadin/addons/ai/formfiller/utils/ComponentUtils.java

```diff
@@ -358,4 +365,29 @@ else if (field.getType().equals(Boolean.class))
 
         grid.setItems(gridItems);
     }
+
+    public static boolean isSupportedComponent(Component component) {
+        return supportedComponentStream().anyMatch(c -> c.equals(component.getClass()));
+    }
+
+    private static Stream<Class<? extends Component>> supportedComponentStream() {
```

Comment:
We discussed in our daily with @mgarciavaadin  and @mshabarov that this part is hard to maintain, but we do not have a better solution right now. And without this static list the logs would be spammed with many messages with child components/layouts, which would hide important log messages.